lib/propolis/src/vmm/mem.rs

```diff
@@ -798,11 +797,14 @@ pub struct MemCtx {
 }
 impl MemCtx {
     /// Reads a generic value from a specified guest address.
-    pub fn read<T: Copy + FromBytes>(&self, addr: GuestAddr) -> Option<T> {
+    pub fn read<T: Copy + FromBytes>(
+        &self,
+        addr: GuestAddr,
+    ) -> Option<GuestData<T>> {
```

Comment:
fwiw this diff misses the same thing i did in #794, which is that `do_read` in `bhyve-api` reads from guest memory too. i was figuring that'd be a relatively small change to chase through this week but i wanted to go read the crate and bhyve side a bit first. if you don't get to it here, that future PR will be me doing "bring `MemCtx::read` improvements to `do_read`", if you do change that here too then that's also fine!

Reply:
My understanding of `do_read` is that it's used to read state from the in-kernel VMM--things like the state of in-kernel emulated devices, bhyve-managed MSRs, and captured VMX/SVM state that might need to be migrated. 
Some of that could technically have come from the guest (via WRMSR or an I/O instruction), but to my mind none of it is really guest *application* data of the sort we're looking to protect in this PR. (I'm going off the list of device state information classes in `bhyve-api/sys/src/vmm_data.rs` to draw this conclusion; there are `VDC_REGISTER` and `VDC_FPU` classes, but AFAICT these haven't actually been implemented for this bhyve interface. Everything else looks to me like internal kernel VMM state.)